lib/train/transports/local.rb

```diff
@@ -52,6 +57,133 @@ def login_command
       def uri
         'local://'
       end
+
+      class WindowsRunner
+        require 'json'
+        require 'base64'
+        require 'securerandom'
+
+        def initialize
+          @pipe = acquire_pipe
+        end
+
+        def run_command(cmd)
+          res = @pipe ? run_via_pipe(cmd) : run_via_shellout(cmd)
```

Comment:
I would split the runner into two, so that we have a `WindowsPipeRunner` and a `WindowsShellRunner`. This would make the implementation more clean. It also allows us later to run the pipe only on supported windows systems.